bench-tps/src/main.rs

```diff
@@ -97,7 +98,7 @@ fn create_connection_cache(
 
     let rpc_client = Arc::new(RpcClient::new_with_commitment(
         json_rpc_url.to_string(),
-        CommitmentConfig::confirmed(),
+        commitment_config,
```

Comment:
When Greg and I looked at this, the hard-coded commitment levels looked suspicious in combination with the `blockhash not found` errors that Greg was seeing:
https://github.com/solana-labs/solana/blob/3284d34e041f737ddb0b946d0695d9c6ada0fc71/bench-tps/src/send_batch.rs#L31-L33

Even though the client is created with one commitment level, we're fetching blockhashes for a different level. I think the problem scenario was:
- RpcClient created with `confirmed`
- RpcClient configured to do `preflight` check
    - This is enabled by default, and this PR is disabling it too
- RpcClient sends transaction using a `processed` blockhash
- RPC node performs preflight simulation using a `Bank` to match the client's commitment level. A `confirmed` `Bank` may not be aware of the `processed` blockhash, thus yielding `blockhash not found`
    - This is seemingly a race condition as the `Bank` and blockhash that were previously `processed` could advance to `confirmed` by the time the `tx` simulation is happening

I think it'd be worth auditing all hard-coded `CommitmentConfig::` instances in `bench-tps`. For example, I'm inclined to think that in the code snippet I linked in this comment, we could just use `client.get_latest_blockhash()`, which would just use the client's cached commitment level. Altho technically, disabling preflight might do it too ?

Reply:
if we change the internals of `get_latest_blockhash()` to call `client.get_latest_blockhash()` then we should probably change the name of the function `get_latest_blockhash()` to something like `get_latest_blockhash_with_client_commitment` or something. since it won't necessarily be the latest blockhash...

Reply:
I also think thanking it to `client.get_latest_blockhash()` would be better than just relying on `skip_preflight`. it is more explicit imo

Reply:
makes sense, I agree that we should use client's confirmation level instead of specifying a custom one.

Reply:
> if we change the internals of get_latest_blockhash() to call client.get_latest_blockhash() then we should probably change the name of the function get_latest_blockhash() to something like get_latest_blockhash_with_client_commitment or something. since it won't necessarily be the latest blockhash..

I disagree with the name change and think we should leave it as-is. The server side function is just `getLatestBlockhash`: https://solana.com/docs/rpc/http/getlatestblockhash

As is the function in `RpcClient`, noting that `get_latest_blockhash_with_commitment()` is deprecated (I'm guessing we might have made it private if we could without potentially breaking people):
https://github.com/solana-labs/solana/blob/139b9c8c2583a12f84f4c49a9f4bfe57a430bcaa/rpc-client/src/nonblocking/rpc_client.rs#L5254-L5262

In addition to retaining consistent naming with these, the commitment is implicitly part of the client and thus the requests you make with the client. So, when you call `getLatestBlockhash`, I think of it as "give me the latest blockhash for the commitment I passed in". To me, you should think about commitment when you create the client, but then not have to worry about it

Reply:
I see. Ok that is fine with me. Didn't realize `get_latest_blockhash_with_commitment()` was deprecated.